### ISSUE TITLE
E305: False positive when using register and returned code

### ISSUE THREAD

**Issue by cedef** [ansible/ansible-lint#388]
# Issue Type: Bug report

# Ansible and Ansible Lint details

```
ansible 2.7.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cedef/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/cedef/.virtualenvs/molecule/local/lib/python2.7/site-packages/ansible
  executable location = /home/cedef/.virtualenvs/molecule/bin/ansible
  python version = 2.7.13 (default, Sep 26 2018, 18:42:22) [GCC 6.3.0 20170516]
ansible-lint 3.4.23
```

- ansible installation method: pip
- ansible-lint installation method: pip

Given the following two tasks:
```
- name: ensure some commands returned 0
  shell: /usr/bin/true
  register: somecmd
  changed_when: false
  tags:
    - bugreport

- name: do something if rc != 0
  command: echo something wrong happen
  when: somecmd.rc != 0
  tags:
    - bugreport
```

# Desired Behaviour

Ansible-lint should not complain about using `shell` module instead of `command` (ANSIBLE0013) because we need `rc` (return code) here.

# Actual Behaviour

```
[ANSIBLE0013] Use shell only when shell functionality is required
/home/cedef/src/cedefinc-ansible/roles/cedefinc/tasks/main.yml:24
Task/Handler: ensure some commands returned 0
```

**Comment by ssbarnea** [comment]
This is not false positive, you should have called "command" instead of shell in this case.
